File: algorithms/strings/PalindromeIndex.java
import java.io.*;
import java.util.*;
import java.text.*;
import java.math.*;
import java.util.regex.*;

// Solution to https://www.hackerrank.com/challenges/palindrome-index

public class Solution {

    public static void main(String[] args) {
        Scanner in = new Scanner(System.in);
        int T = in.nextInt();
        for(int i = 0; i < T; i++){
            String s = in.next();
            int l = 0;
            int r = s.length() - 1;
            int index = -1;
            while(l < r){
                if(s.charAt(l) != s.charAt(r)){
                    String left = s.substring(l, r);
                    String right = s.substring(l+1, r+1);
                    if(left.equals(new StringBuilder(left).reverse().toString())){
                        index = r;
                        break;
                    }else if(right.equals(new StringBuilder(right).reverse().toString())){
                       index = l;
                       break;
                    } else{
                        index = -1;
                        break;
                    }
                }else{
                    l++;
                    r--;
                }
            }
            System.out.println(index);
        }
    }
}

File: algorithms/strings/palindrome_index.py
# Solution to https://www.hackerrank.com/challenges/palindrome-index

t = int(input())
for _ in range(t):
    s = input()
    start = 0
    end = len(s) - 1
    ans = -1
    while start < end:
        if s[start] != s[end]:
            left = s[start:end]
            right = s[start+1:end+1]
            if left == left[::-1]:
                ans = end
            elif right == right[::-1]:
                ans = start
            else:
                ans = -1
            break
        start = start + 1
        end = end - 1
    print(ans)
